bazel/external/vpp_vcl.genrule_cmd

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+# This works only on Linux.
+if [[ `uname` != "Linux" ]]; then
+  echo "ERROR: VPP VCL is currently supported only on Linux"
+  exit 1
+fi
+
+# Bazel magic.
+ROOT=$$(dirname $(rootpath src/CMakeLists.txt))/..
+
+echo "ROOT $$ROOT"
+pushd $$ROOT
+
+# Handle extra dependencies
+pip3 install ply
+
+echo $$(cmake --version | head -1)
+echo "ninja version" $$(ninja --version)
+
+# Build
+mkdir _vcl
+cd _vcl
+cmake -G Ninja ../src -DCMAKE_BUILD_TYPE:STRING=release
```

Comment:
Another issue was the patching of the build system and the script above that statically builds VCL (leveraging cmake to build a target instead of the whole thing) and copies files from weird locations. If we support all that, and the python dep, as part of envoy_cmake, I can give it a try.

Reply:
Just gave `foreign_cc/BUILD` a quick read and as far as I can tell `envoy_cmake` doesn't accept custom targets for ninja, i.e., it can't build a subset of a given project. So replicating this genurule_cmd doesn't seem possible.

Reply:
> Don't we use requirements.txt for managing python deps? @phlax

apologies i missed this - yes - i think this dep should be managed by bazel/pip/requirements

the problem you will face is that adding a `requirement(ply)` to a genrule's tools will not set up the `PYTHONPATH` automatically

you could hardcode the `PYTHONPATH` which would workaround this issue, but its a pretty ugly hack

what would be much easier is to set the parts that require the python dependency in a `py_binary` which in turn should depend on the python/pip requirement

i can help doing this if necessary

Reply:
Hi @phlax! I guess I've hit the issue above while trying to move the `ply` dependency to `repositories_extra.bzl` :-)

Not sure I completely understood the solution above, but for context: this genrule configures [vpp](https://github.com/FDio/vpp) but only ninja builds the vppcom [target](https://github.com/FDio/vpp/blob/master/src/vcl/CMakeLists.txt) (vcl). Because vcl has a dependency on the api layer, [this](https://github.com/FDio/vpp/blob/master/src/cmake/api.cmake#L41) script that generates the api files needs to be executed, hence the dependency on `ply`. Can that be made a `py_binary`?

Reply:
so im assuming that whatever is called just needs to have `ply` in `PYTHONPATH` so python can make use of it

my thought is to move (some of) the shell script into a `py_binary` and to use `subprocess.run` to run any shell commands

that way the `py_binary` can depend on the `requirement` and bring it into the path.

Reply:
not sure about using `repositories_extra.bzl` - i dont think that should be necessary

you need to add `ply` to `tools/base/requirements.in` and then follow the instructions at the top of the adjacent `requirements.txt` file (run it in the same directory to minimize changes)

then you can use 

```
load("@base_pip3//:requirements.bzl", "requirement")
```

and add

```
deps = [requirement("ply")],
```

to the `py_binary`

Reply:
Yup! Also, that dependency is needed by one script used during the build bootstrap but not afterwards. Do we have any example of such a python script to use as a template? Or alternatively, where should we add that script?

Reply:
hmm, good q, not sure - i tend to put all python tools in the `/tools/` dir but probs in this case i would say in the same directory as where you have the sh script now